File: BO.Core/Models/ShowListResponseModel.cs
using System;
using System.Collections.Generic;

namespace BO.Core.Models
{
    public class ShowListResponseModel
    {
        public List<ShowItemModel> Items { get; set; }
        public int TotalItems { get; set; }
    }

    public class ShowItemModel
    {
        public string Name { get; set; }
        public List<SessionItemModel> Sessions { get; set; }
    }

    public class SessionItemModel
    {
        public Guid Id { get; set; }
        public int FreeSeats { get; set; }
        public DateTimeOffset From { get; set; }
        public DateTimeOffset To { get; set; }
    }
}

File: BO.Web/ClientApp/src/Shared/Models/ShowListResponseModel.ts
export class ShowListResponseModel {
    items: ShowItemModel[] | undefined;
    totalItems: number | undefined;
}

class ShowItemModel {
    name: string | undefined;
    sessions: SessionItemModel[] | undefined;
}

class SessionItemModel {
    id: string | undefined;
    freeSeats: number | undefined;
    from: string | undefined;
    to: string | undefined;
}